src/citrine/resources/material_run.py

```diff
@@ -166,3 +167,54 @@ def get_history(self, scope, id) -> Type[MaterialRun]:
         model = loads(json.dumps([data['context'], data['root']], cls=TaurusEncoder))
         # Convert taurus objects into citrine-python objects
         return MaterialRun.build(model)
+
+    def with_spec(self,
+                  spec_id: str,
+                  spec_scope: str = 'id',
+                  per_page: int = 20) -> Iterator[dict]:
+        """
+        [ALPHA] Get all material runs associated with a material spec.
+
+        The material spec is specified by its scope and id.
+
+        :param spec_id: The unique id corresponding to `scope`.
+            The lookup will be most efficient if you use the Citrine ID (scope='id')
+            of the material spec.
+        :param spec_scope: The scope used to locate the material spec.
+        :param per_page: The number of results to return per page.
+        :return: A search result of material runs
+        """
+        path_prefix = MaterialSpecCollection(self.project_id,
+                                             self.dataset_id,
+                                             self.session)._get_path(ignore_dataset=True)
+        path = path_prefix + "/" + spec_scope + "/" + spec_id + "/material-runs"
+        return self.session.cursor_paged_resource(self.session.get_resource,
+                                                  path,
+                                                  per_page=per_page,
+                                                  version="v1")
+
+    # Retrieve all material runs associated with a material template
+    def with_template(self,
```

Comment:
Thoughts on calling this `filter_by_template`?

Reply:
I am on the side of keeping this consistent by using filter_by_template.